Exporting trubank store fails

I think what's happening is when we're iterating through the trunbank store, we are unmarshalling the `UserList` into a `Transaction`.

```
panic: Not enough bytes to read in UnmarshalBinaryLengthPrefixed, want 206 more bytes but only have 0

goroutine 1 [running]:
github.com/TruStory/truchain/vendor/github.com/tendermint/go-amino.(*Codec).MustUnmarshalBinaryLengthPrefixed(...)
    /home/ubuntu/go/src/github.com/TruStory/truchain/vendor/github.com/tendermint/go-amino/amino.go:316
github.com/TruStory/truchain/x/trubank.ExportGenesis.func1(0xc000c96fb8, 0x2, 0x2, 0x2)
    /home/ubuntu/go/src/github.com/TruStory/truchain/x/trubank/genesis.go:24 +0x1cb
github.com/TruStory/truchain/types.Keeper.EachPrefix(0xc0004a6540, 0x14a9680, 0xc0007f7c00, 0x14b9120, 0xc000807d40, 0xc0008c2940, 0xd, 0x0, 0x0, 0xc0013f0998, ...)
    /home/ubuntu/go/src/github.com/TruStory/truchain/types/keeper.go:93 +0x14c
github.com/TruStory/truchain/types.Keeper.Each(...)
    /home/ubuntu/go/src/github.com/TruStory/truchain/types/keeper.go:106
github.com/TruStory/truchain/x/trubank.ExportGenesis(0x14b9120, 0xc000807d40, 0xc0008c2940, 0xd, 0xc0004a6540, 0x14a9680, 0xc0007f7c00, 0x14cc320, 0xc00000c780, 0x14c3980, ...)
    /home/ubuntu/go/src/github.com/TruStory/truchain/x/trubank/genesis.go:22 +0x159
```

--- a/types/keeper.go
+++ b/types/keeper.go
@@ -108,10 +108,10 @@ func (k Keeper) Each(ctx sdk.Context, fn func([]byte) bool) (err sdk.Error) {
 
 // GetIDKey returns the key for a given index
 func (k Keeper) GetIDKey(id int64) []byte {
-	return []byte(fmt.Sprintf("%s:id:%d", k.GetStoreKey().Name(), id))
+	return []byte(fmt.Sprintf("%s%d", k.StorePrefix(), id))
 }
 
 // StorePrefix returns the root prefix of the key-value store
-func (k Keeper) StorePrefix(storeKey string) string {
-	return fmt.Sprintf("%s:id:", storeKey)
+func (k Keeper) StorePrefix() string {
+	return fmt.Sprintf("%s:id:", k.GetStoreKey().Name())
 }

--- a/x/backing/genesis.go
+++ b/x/backing/genesis.go
@@ -26,7 +26,7 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
 // ExportGenesis exports the genesis state
 func ExportGenesis(ctx sdk.Context, keeper Keeper) GenesisState {
 	var backings []Backing
-	err := keeper.Each(ctx, func(bz []byte) bool {
+	err := keeper.EachPrefix(ctx, keeper.StorePrefix(), func(bz []byte) bool {
 		var b Backing
 		keeper.GetCodec().MustUnmarshalBinaryLengthPrefixed(bz, &b)
 		backings = append(backings, b)

--- a/x/challenge/genesis.go
+++ b/x/challenge/genesis.go
@@ -30,7 +30,7 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
 // ExportGenesis exports the genesis state
 func ExportGenesis(ctx sdk.Context, keeper Keeper) GenesisState {
 	var challenges []Challenge
-	err := keeper.Each(ctx, func(bz []byte) bool {
+	err := keeper.EachPrefix(ctx, keeper.StorePrefix(), func(bz []byte) bool {
 		var c Challenge
 		keeper.GetCodec().MustUnmarshalBinaryLengthPrefixed(bz, &c)
 		challenges = append(challenges, c)

--- a/x/trubank/genesis.go
+++ b/x/trubank/genesis.go
@@ -21,8 +21,7 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
 // ExportGenesis exports the genesis state
 func ExportGenesis(ctx sdk.Context, keeper Keeper) GenesisState {
 	transactions := make([]Transaction, 0)
-	prefix := keeper.StorePrefix(StoreKey)
-	err := keeper.EachPrefix(ctx, prefix, func(bz []byte) bool {
+	err := keeper.EachPrefix(ctx, keeper.StorePrefix(), func(bz []byte) bool {
 		var tx Transaction
 		keeper.GetCodec().MustUnmarshalBinaryLengthPrefixed(bz, &tx)
 		transactions = append(transactions, tx)

